Automattic/jetpack
# Instagram Gallery Block: Add fixture and unit tests

Adds unit and fixture tests for the Instagram Gallery Block (also known as the Latests Instagram Posts block). The block is refactored slightly to move the controls to a separate component to get it under test.

Much of the Instagram connection logic is mocked in the provided tests, so that the tests can focus on the UI. I've skipped testing parts of the Instagram and wpcom connection process that would be difficult to get under test due to usage of APIs or of parts of Gutenberg redux state. We can always follow up with additional tests in future PRs if we'd like to address that in greater detail.

#### Changes proposed in this Pull Request:

* Add unit and fixture tests for the Instagram Gallery Block
* Remove `isLarge` prop from the buttons as it's no longer used by the Button block and is currently throwing a warning in the console
* Refactor the block's controls so that they're in a separate component

#### Jetpack product discussion


See p1HpG7-b3G-p2

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Sync to wpcom, insert a Latest Instagram Posts block and connect an Instagram account.
* Ensure the block behaves as expected

Ensure tests pass:

* Change directory to `cd projects/plugins/jetpack` and run the following, making sure all the tests pass:
* `yarn jest extensions/blocks/instagram-gallery/test`
